File: julia/algorithms/epsilon_greedy/test_annealing.jl
load("core.jl")

srand(1)
means = [0.1, 0.1, 0.1, 0.1, 0.9]
n_arms = length(means)
means = means[randperm(n_arms)]
arms = map(mu -> BernoulliArm(mu), means)
println("Best arm is $(ind_max(means))")

total_results = zeros(0, 5)

algo = AnnealingEpsilonGreedy(zeros(Int64, n_arms), zeros(n_arms))
results = test_algorithm(algo, arms, 5000, 250)
total_results = vcat(total_results, results)

csvwrite("algorithms/epsilon_greedy/annealing_results.csv", total_results)

File: python/algorithms/epsilon_greedy/test_annealing.py
execfile("core.py")

import random

random.seed(1)
means = [0.1, 0.1, 0.1, 0.1, 0.9]
n_arms = len(means)
random.shuffle(means)
arms = [BernoulliArm(m) for m in means]
print("Best arm is " + str(ind_max(means)))

my_algo = AnnealingEpsilonGreedy([], [])
my_algo.initialize(n_arms)
results = test_algorithm(my_algo, arms, 5000, 250)

with open("algorithms/epsilon_greedy/annealing_results.tsv", "w") as f:
    for i in range(len(results[0])):
        f.write("\t".join(str(res[i]) for res in results) + "\n")
